crates/esthri/src/blocking.rs

```diff
@@ -19,24 +19,15 @@ use crate::Result;
 use crate::S3PathParam;
 
 #[tokio::main]
-pub async fn head_object<T>(
-    s3: &T,
-    bucket: impl AsRef<str>,
```

Comment:
impl AsRef<str> is used intentionally throughout the public API to make things more flexible as the call site-- this allows user's to mix String and &str for example